src/schedule.rs

```diff
@@ -1,18 +1,22 @@
-use crate::error::{Error, ErrorKind};
+
 use chrono::offset::TimeZone;
-use chrono::{DateTime, Datelike, Duration, Timelike, Utc};
-use nom::{types::CompleteStr as Input, *};
-use std::collections::BTreeSet;
+use chrono::{DateTime, Datelike, Utc};
 use std::collections::Bound::{Included, Unbounded};
 use std::fmt::{Display, Formatter, Result as FmtResult};
-use std::iter::{self, Iterator};
-use std::str::{self, FromStr};
 
 use crate::time_unit::*;
+use crate::ordinal::*;
+use crate::queries::*;
+
+impl From<Schedule> for String {
+    fn from(schedule: Schedule) -> String {
+        schedule.source.unwrap()
+    }
+}
```

Comment:
The [docs for the `From` trait](https://doc.rust-lang.org/std/convert/trait.From.html) say:

> **Note: This trait must not fail.** If the conversion can fail, use [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html).

Given that `unwrap()` will panic if `source` is not a valid cron expression, I don't think we should provide a `From` impl. We could provide a `TryFrom<&str>`, but that's very close to `FromStr`. Do you have a specific use case for this?

For now, I'd leave this out of this PR so we can avoid renaming `Schedule::from`. You mentioned that this is your first time contributing to another project; I suggest doing most changes as new commits that you push to the same branch instead of force pushing new versions. This lets me review each new diff individually and preserves any comments/conversation we've added to the review. (Force pushing will discard comments around the code since it doesn't know what goes with what after the rewrite.)

Reply:
>Given that unwrap() will panic if source is not a valid cron expression, I don't think we should provide a From impl. We could provide a TryFrom<&str>, but that's very close to FromStr. Do you have a specific use case for this?
Yeah no worries, I can do that! It's just code I copied though. I didn't add it.

Also as you said in: https://github.com/zslayton/cron/pull/62#issuecomment-768288060 source is always available, so this won't ever panic.

#71 would also fix this.

Reply:
🤦 I read it backwards. I thought this was going from `String` -> `Schedule` and calling `unwrap()` on the parsing result. Hadn't had enough coffee yet, I guess.

I'm ok with having a `From<Schedule>` impl! Let's save that change for the next PR with the `ScheduleFields` change, we can rename the `from` constructor to `new`.

Reply:
Yeah it was already in the schedule.rs file, I only copied it to the parsing.rs file. I can keep it in schedule.rs and make it public/expose if you prefer?

Also... I just smashed my laptop screen. It's getting repaired tomorrow. I'm not sure if I can use my computer the next couple of weeks. I got a couple of branches that I wanted to create a pull request for (way smaller than this one). Is that ok?

Reply:
> Also... I just smashed my laptop screen.

Eep! I'm sorry to hear that!

> It's getting repaired tomorrow. I'm not sure if I can use my computer the next couple of weeks. I got a couple of branches that I wanted to create a pull request for (way smaller than this one). Is that ok?

Yep, go for it. I've gone ahead and merged this one, we can always iron out any wrinkles we notice in another one.

Good luck with your laptop and thank you for contributing!